tinygrad/tensor.py

```diff
@@ -208,9 +208,7 @@ def swish(self):
     return self.mul(self.sigmoid())
 
   def tanh(self):
-    t2 = Tensor(np.zeros(self.shape, dtype=self.dtype)+2, gpu=self.gpu, requires_grad=False)
-    t1 = Tensor(np.zeros(self.shape, dtype=self.dtype)+1, gpu=self.gpu, requires_grad=False)
-    return self.mul(t2).sigmoid().mul(t2) - t1 # 2*sigmoid(2*x)-1
+    return 2.0 * ((2.0 * self).sigmoid())  - 1.0 # 2*sigmoid(2*x)-1
```

Comment:
I see 3 lines to save in the other ops!

Reply:
@geohot i saved them in mean, sqrt but got error in gpuops :(
can u help me debug it.

Reply:
Hmm, I'm not sure tanh is tested...paste the error

Reply:
tanh is tested

Reply:
E     TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
E         1. pyopencl._cl.Buffer(context: pyopencl._cl.Context, flags: int, size: int = 0, hostbuf: object = None)
E     
E     Invoked with: <pyopencl.Context at 0x557e1da84a40 on <pyopencl.Device 'GeForce GTX 1650' on 'NVIDIA CUDA' at 0x557e1e3d09a0>>, 33, 4.0; kwargs: hostbuf=array([1.], dtype=float32)

tinygrad/tensor.py:45: TypeError

Reply:
when changed
```
def mean(self)
   return self.sum() * (1/np.prod(self.shape))


```

Reply:
@geohot  this is solved :)